pkg/controller/ovsnodeosp/ovsnodeosp_controller.go

```diff
@@ -0,0 +1,323 @@
+package ovsnodeosp
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/openstack-k8s-operators/neutron-operator/pkg/common"
+
+	"github.com/openstack-k8s-operators/neutron-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+
+	neutronv1 "github.com/openstack-k8s-operators/neutron-operator/pkg/apis/neutron/v1"
+	ovsnodeosp "github.com/openstack-k8s-operators/neutron-operator/pkg/ovsnodeosp"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_ovsnodeosp")
+
+// Add creates a new OVSNodeOsp Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileOVSNodeOsp{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("ovsnodeosp-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource OVSNodeOsp
+	err = c.Watch(&source.Kind{Type: &neutronv1.OVSNodeOsp{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch ConfigMaps owned by OVSNodeOsp
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		IsController: false,
+		OwnerType:    &neutronv1.OVSNodeOsp{},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner OVSNodeOsp
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &neutronv1.OVSNodeOsp{},
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
Perhaps missing here do we need to watch for changes to the DaemonSet's too right?